SimPEG/Problem.py

```diff
@@ -123,10 +123,11 @@ def unpair(self):
     deleteTheseOnModelUpdate = []
 
     @properties.observer('model')
-    def _on_model_update(self, value):
-        for prop in self.deleteTheseOnModelUpdate:
-            if hasattr(self, prop):
-                delattr(self, prop)
+    def _on_model_update(self, change):
+        if self.model is None or not np.allclose(self.model, change['value']):
```

Comment:
I wonder if an early exit of `self.model is change['value']` would be faster. If the model is large a point wise comparison is going to be slow. If they are pointing to the same block in memory, then we can skip that.